promql/prettier/prettier.go

```diff
@@ -475,6 +480,19 @@ func (p *Prettier) parseFile(name string) (*rulefmt.RuleGroups, []error) {
 	return groups, nil
 }
 
+// isMetricNameAtomic returns true if the metric name is singular in nature.
+// This is used during sorting of lex items.
+func isMetricNameNotAtomic(metricName string) bool {
+	// We aim to convert __name__ to an ideal metric_name if the value of that labels is atomic.
+	// Since a non-atomic metric_name will contain alphabets other than a-z and A-Z including _,
+	// anything that violates this ceases the formatting of that particular label item.
+	// If this is not done then the output from the prettier might be an un-parsable expression.
+	if regexp.MustCompile("^[a-z_A-Z]+$").MatchString(metricName) {
```

Comment:
Certain keywords, like `bool`, also can't be metric names. This is documented in the PromQL docs.

Reply:
Sure. Will implement checks for that.